packages/commonwealth/client/scripts/helpers/abi_form_helpers.ts

```diff
@@ -0,0 +1,71 @@
+import Web3 from 'web3';
+import { AbiInput, AbiItem } from 'web3-utils';
+import { BigNumber, ethers } from 'ethers';
+import { ValidationStatus } from '../views/components/component_kit/cw_validation_text';
+
+const Bytes32 = ethers.utils.formatBytes32String;
+
+export function encodeFunctionSignature(abi: AbiItem): string {
+  const coder = new Web3().eth.abi;
+  return coder.encodeFunctionSignature(abi);
+}
+
+export function validateAbiInput(
+  val: string,
+  inputType: string
+): [ValidationStatus, string] {
+  const coder = new Web3().eth.abi;
+  try {
+    if (inputType.slice(-2) === '[]') {
+      coder.encodeParameter(inputType, JSON.parse(val));
+    } else {
+      if (inputType === 'bool') {
+        if (val !== 'true' && val !== 'false') {
+          return ['failure', 'Input must be a boolean'];
+        }
+      }
+      coder.encodeParameter(inputType, val);
+    }
+    return ['success', ''];
+  } catch (e) {
+    return ['failure', e.message];
+  }
+}
+
+export function handleMappingAbiInputs(
+  inputIndex: number,
+  input: string,
+  functionName: string,
+  inputMap: Map<string, Map<number, string>>
+) {
+  if (!inputMap.has(functionName)) {
+    inputMap.set(functionName, new Map<number, string>());
+    const inputArgMap = inputMap.get(functionName);
+    inputArgMap.set(inputIndex, input);
+    inputMap.set(functionName, inputArgMap);
+  } else {
+    const inputArgMap = inputMap.get(functionName);
+    inputArgMap.set(inputIndex, input);
+    inputMap.set(functionName, inputArgMap);
+  }
+}
+
+export function processAbiInputsToDataTypes(
```

Comment:
Search to see if web3 library does auto parsing

Reply:
could not find auto parser, but found great abi-decoding library. Will have to do refactors, but should remove a helper function we use. If we use this https://github.com/ConsenSys/abi-decoder

Reply:
moved this into a different branch to abstract this refactor

Reply:
btw the flaw with the Consensys Abi Decoder is you need to have the Exact abi of the contract you are calling because the encoding has to be exactly the same, which takes away from use cases like using an interface to interact with a contract. So I am going to leave this out for now